pkg/volume/csi/csi_client_test.go

```diff
@@ -529,30 +596,78 @@ func TestClientNodeStageVolume(t *testing.T) {
 	testPath := filepath.Join(tmpDir, "/test/path")
 
 	testCases := []struct {
-		name              string
-		volID             string
-		stagingTargetPath string
-		fsType            string
-		secrets           map[string]string
-		mountOptions      []string
-		mustFail          bool
-		err               error
+		name                           string
+		volID                          string
+		stagingTargetPath              string
+		fsType                         string
+		secrets                        map[string]string
+		mountOptions                   []string
+		fsGroup                        *int64
+		delegateFSGroupFeatureGate     bool
+		driverSupportsVolumeMountGroup bool
+		expectedFSGroupInNodeStage     string
+		mustFail                       bool
+		err                            error
 	}{
 		{name: "test ok", volID: "vol-test", stagingTargetPath: testPath, fsType: "ext4", mountOptions: []string{"unvalidated"}},
 		{name: "missing volID", stagingTargetPath: testPath, mustFail: true},
 		{name: "missing target path", volID: "vol-test", mustFail: true},
 		{name: "bad fs", volID: "vol-test", stagingTargetPath: testPath, fsType: "badfs", mustFail: true},
 		{name: "grpc error", volID: "vol-test", stagingTargetPath: testPath, mustFail: true, err: errors.New("grpc error")},
+
+		{
+			name:                           "fsgroup provided, DelegateFSGroupToCSIDriver feature enabled, driver supports volume mount group; expect fsgroup to be passed to NodePublishVolume",
+			volID:                          "vol-test",
+			stagingTargetPath:              testPath,
+			fsType:                         "ext4",
+			fsGroup:                        &testFSGroup,
+			delegateFSGroupFeatureGate:     true,
+			driverSupportsVolumeMountGroup: true,
+			expectedFSGroupInNodeStage:     "3000",
+		},
+		{
+			name:                           "fsgroup not provided, DelegateFSGroupToCSIDriver feature enabled, driver supports volume mount group; expect fsgroup not to be passed to NodePublishVolume",
+			volID:                          "vol-test",
```

Comment:
these tests say `NodePublishVolume`, but we are checking the case for `NodeStage`

Reply:
Thanks for the catch